src/com/magento/idea/magento2plugin/actions/generation/dialog/NewControllerDialog.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+package com.magento.idea.magento2plugin.actions.generation.dialog;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiDirectory;
+import com.intellij.psi.PsiFile;
+import com.magento.idea.magento2plugin.actions.generation.NewControllerAction;
+import com.magento.idea.magento2plugin.actions.generation.data.ControllerFileData;
+import com.magento.idea.magento2plugin.actions.generation.dialog.validator.NewControllerValidator;
+import com.magento.idea.magento2plugin.actions.generation.generator.ModuleControllerClassGenerator;
+import com.magento.idea.magento2plugin.magento.files.ControllerPhp;
+import com.magento.idea.magento2plugin.magento.packages.Package;
+import com.magento.idea.magento2plugin.ui.FilteredComboBox;
+import com.magento.idea.magento2plugin.util.magento.GetModuleNameByDirectory;
+
+import javax.swing.*;
+import java.awt.event.*;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class NewControllerDialog extends AbstractDialog {
+    public static String ADMINHTML_AREA = "adminhtml";
+    public static String FRONTEND_AREA = "frontend";
```

Comment:
We already have `com.magento.idea.magento2plugin.magento.packages.Package.Areas#adminhtml` and `com.magento.idea.magento2plugin.magento.packages.Package.Areas#frontend`